### PR TITLE
GEODE-3977: Modifying the QueryService template

### PR DESCRIPTION
**Pull request by mhansonp**
-Modified the QueryService template to only take
 one parameter instead of two.

### CODE REVIEW

File: clicache/src/QueryService.hpp [hunk 1]
```diff
@@ -51,7 +60,7 @@ namespace Apache
       /// <summary>
       /// Provides a query service.
       /// </summary>
-      generic<class TKey, class TResult>
+      generic<class TResult>
```
Comment (pivotal-jbarrett) [inline clicache/src/QueryService.hpp]: The more I think about it does it even make sense for the class itself to be templated. Does the class itself depend in any way on the `TResult`. I think the answer is no if you look at each method. The bulk of the methods are factories for `Query<TResult>` and `CqQuery<TKey, TResult>`.  So should have to ask for a new `QueryService` every time I want to make new `Query` with a different `TResult`? That seems wrong to me.
```
var qsString = cache.getQueryService<string>();
var qString = qsString.createQuery("....");
var qsInt = cache.getQueryService<int>)();
var qInt = qsInt.createQuery("...");
```
vs.
```
var qs = cache.getQueryService();
var qString = qs.createQuery<string>("....");
var qInt = qs.createQuery<int>("...");
```

Comment (pivotal-jbarrett) [reply clicache/src/QueryService.hpp]: Additionally methods like `CloseCqs` operates on all CQs, not just the ones that match the `TResult` of the class.

File: clicache/src/QueryService.hpp [hunk 2]
```diff
@@ -66,65 +75,66 @@ namespace Apache
...
 
         /// @nativeclient
         /// <summary>
         /// Get all  <c>CqQuery</c> on this client.
         /// </summary>
         /// @endnativeclient
-  //generic<class TKey, class TResult>
-	array<CqQuery<TKey, TResult>^>^ GetCqs();
+        generic<class TKey>
+	      array<CqQuery<TKey, TResult>^>^ GetCqs();
```
Comment (pivotal-jbarrett) [inline clicache/src/QueryService.hpp]: Formatting is getting out of whack here.

File: clicache/src/QueryService.hpp (hunk 1 above)
Comment (mhansonp) [reply clicache/src/QueryService.hpp]: I agree, so I will pull all of the templating out of QueryService.

**Pull request closed without merging**